pennylane/optimize/shot_adaptive.py

```diff
@@ -338,6 +410,20 @@ def _single_shot_qnode_gradients(self, qnode, args, kwargs):
         if self.lipschitz is None:
             self.check_learning_rate(1)
 
+        qnode.construct(args, kwargs)
+        [expval] = qnode.tape.measurements
+        if self.term_sampling == "weighted_random_sampling":
+            if isinstance(expval.obs, qml.Hamiltonian):

```

Comment:
> On the other hand, we can change the default value to None since that's what it has effectively been for QNodes. This would be "breaking" for ExpvalCost users, but surely there are none left given the amount of time the deprecation warning has been present for.

This works for me, i.e.,
- `term_sampling=None` by default
- adding a breaking change in the changelog
- Can we also additionally raise a warning in `ShotAdaptiveOptimizer` if an `ExpvalCost` is used, recommending users switch to using `qml.expval(H)` as well as setting `term_sampling='weighted_random_sampling'` - I can help with the warning message here.

Reply:
Can we also update the docstrings, e.g., line 32, 39, 50, to mention these updates.

Reply:
I'd rather not raise warnings in the code, because we only see the type in `opt.step` which might be called many, many times by users.

we already have a [warning in the ExpvalCost doc](https://docs.pennylane.ai/en/stable/code/api/pennylane.ExpvalCost.html), and we [raise a warning](https://github.com/PennyLaneAI/pennylane/blob/b1f68fb9d800aa83a8d81134127a7d6cd7d6644b/pennylane/vqe/vqe.py#L155-L160) when someone creates an `ExpvalCost` instance.

I'll add a similar warning to the `ShotAdaptiveOptimizer` docs